Automattic/wp-calypso
# Extensions: Add extension reducers to static build

This commit adds a reducer from any extention that has a
`state/reducer.js` file. It builds this in statically to the primary
reducer chain and runs it for all actions, just like any other reducer
in Calypso.

This is patterned off of PR #8103 from the original Calypso Extension
work.

It also adds a very simple example of using this state in the
`hello-dolly` example extension. Review of the example code was
performed already in PR #11347

To Test:
1. Run `make test` to verify all tests are passing.
2. Run `make run`
3. Visit `http://calypso.localhost:3000/hello-dolly`
4. Verify that the lyrics show up in bold.
5. Click 'Next' to see the next lyric.
